src/library/scala/collection/Seq.scala

```diff
@@ -348,13 +369,22 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     *  @return  the index `<= end` of the last element of this $coll that satisfies the predicate `p`,
     *           or `-1`, if none exists.
     */
-  def lastIndexWhere(p: A => Boolean, end: Int = length - 1): Int = {
+  def lastIndexWhere(p: A => Boolean, end: Int): Int = {
     var i = length - 1
     val it = reverseIterator
     while (it.hasNext && { val elem = it.next(); (i > end || !p(elem)) }) i -= 1
     i
   }
 
+  /** Finds index of last element satisfying some predicate.
+    *
+    *  @param   p     the predicate used to test elements.
+    *  @return  the index of the last element of this $coll that satisfies the predicate `p`,
+    *           or `-1`, if none exists.
+    */
+  @deprecatedOverriding("Override lastIndexWhere(p, end) instead - lastIndexWhere(p) calls lastIndexWhere(p, length-1)", "2.13.0")
```

Comment:
It's not ideal to have `length` here; `List` will need to traverse itself completely in order to compute the `length` and that value won't help it at all because it can't count on it so will have to do it all over again in the method.  Can we use `Int.MaxValue` instead?  The `lastIndexWhere` and the like methods should all handle unnecessarily generous last values anyway.

Reply:
Hm, that might have been the reason why those methods weren't `final`. I'll change it to `MaxValue`.